`make install` build error in libp2p-noise dep

`make install` fails with (reproduced on mac on linux):

```shell
error[E0282]: type annotations needed

   --> /usr/local/rust/cargo/registry/src/github.com-1ecc6299db9ec823/libp2p-noise-0.31.0/src/protocol/x25519.rs:221:45
    |
221 |         curve25519_sk.copy_from_slice(&hash.as_ref()[..32]);
    |                                        -----^^^^^^--
    |                                        |    |
    |                                        |    cannot infer type for type parameter `T` declared on the trait `AsRef`
    |                                        this method call resolves to `&T`
    |
    = note: type must be known at this point
```

--- a/.changelog/unreleased/bug-fixes/908-libp2p-noise-build-patch.md
+++ b/.changelog/unreleased/bug-fixes/908-libp2p-noise-build-patch.md
@@ -1,0 +1,3 @@
+- Dependency: Backport libp2p-noise patch that fixes a compilation
+  issue from <https://github.com/libp2p/rust-libp2p/pull/2264>
+  ([#908](https://github.com/anoma/anoma/issues/908))

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,6 +24,8 @@ exclude = [
 tracing = {git = "https://github.com/tokio-rs/tracing/", rev = "6cc6c47354ceeb47da7c95faa41c6d29b71b5f37"}
 tracing-core = {git = "https://github.com/tokio-rs/tracing/", rev = "6cc6c47354ceeb47da7c95faa41c6d29b71b5f37"}
 tracing-subscriber = {git = "https://github.com/tokio-rs/tracing/", rev = "6cc6c47354ceeb47da7c95faa41c6d29b71b5f37"}
+# TODO backported patch in the noise protocl for <https://github.com/libp2p/rust-libp2p/pull/2264>, blocked on libp2p upgrade <https://github.com/anoma/anoma/pull/751>
+libp2p = {git = "https://github.com/heliaxdev/rust-libp2p.git", rev = "1abe349c231eb307d3dbe03f3ffffc6cf5e9084d"}
 
 [profile.release]
 lto = true

